### DISCUSSION TITLE
updating Lyon after finding out that many recordings were misattributed (in progress)

### DISCUSSION THREAD

**Discussion by alecristia** [LAAC-LSCP/ChildProject#309] [Show and tell]
We found out that many of the its and wavs associated to specific children-days were wrong, so Lyon needs a major rehaul. In this S&T I'll explain my reasoning in the early stages of correction -- namely correcting the .its, audio recordings, vtc, and metadata. I'm not going to cover fixing of the human annotations.

- navigate to my local version of the repo & activate environment
- create a branch `git checkout -b newrecs`
- open `scripts/creation.sh` to take notes there[^1]
- remove all the old wavs and annotations

[^1]: I could have created a new script, but I thought that would be even more confusing, since after this rehaul, there will be basically no trace of the old created dataset

```bash
child-project remove-annotations . --set its --recursive
child-project remove-annotations . --set vtc --recursive
child-project remove-annotations . --set reprocessed_its --recursive
child-project remove-annotations . --set cha --recursive

rm annotations/its/raw/*
rm -r annotations/reprocessed_its
rm annotations/vtc/raw/*
rm -r annotations/cha/lena

```

- copied over all the raw wavs and annotations

```bash
datalad unlock annotations/
mkdir -p annotations/its/raw/
cp ~/../../Volumes/Seagate\ Expansion\ Drive/its/*.its annotations/its/raw/.

rm -r recordings/originals/
rm recordings/raw/*
mkdir -p recordings/trs/
cp ~/../../Volumes/Seagate\ Expansion\ Drive/its/*.wav recordings/raw/.
cp ~/../../Volumes/Seagate\ Expansion\ Drive/its/*.trs recordings/trs/.

mkdir -p recordings/upl/
cp ~/../../Volumes/Lyon_Lena/upls/*.upl recordings/upl/.

```

I haven't been able to correctly do the metadata, so I cannot yet import annotations. Also missing is analyses using vtc, to be done on oberon, once the data are downloaded over there.

**Comment by alecristia** [comment]
next, I fixed the metadata

```bash

# working on the metadata
# I took metadata.py from bergelson (within EL1000)
mkdir -p metadata/original/
cp /Users/valentinthouzeau/Documents/projets/SES_language/data_EL1000/lyon/lyon_metadata.csv metadata/original/.
# 20211018 we had generated a new base for the metadata in extra/errors/lyon_EL1000_metadata.csv
# it didn't have all the needed info, so created a script to generate it:
python scripts/fix_metadata.py

# then adapted metadata.py
pip install git+ssh://git@gin.g-node.org:/EL1000/tools.git --upgrade
python scripts/metadata.py lyon extra/errors/lyon_EL1000_metadata_sibs.csv

child-project compute-durations .

datalad save . -m "metadata"
datalad push

```



```
#next import its
# I took import.py from bergelson (within EL1000)
python scripts/import.py its

child-project validate . --annotations its

datalad save . -m "import annotations, check fails because its import was split but not audios"
datalad push --to gin
```

still left to double check
- recording dates
- recording start time
- fixing start time summer time
- recording duration

still left to do
- audio conversion
- vtc
- splitting of sound files??
- import textgrid/cha, import eaf 2021 correct naming if need be, calculate reliability

**Comment by alecristia** [comment]
To do the next batch of actions, it's better to be on the cluster, so I did `datalad get recordings/raw -J 8` there. Output looks good globally:

>
get(ok): recordings/raw/e20130311_153528_008344.wav (file) [from origin...]
get(ok): recordings/raw/e20120918_153122_008066.wav (file) [from origin...]
get(ok): recordings/raw/e20130604_135718_008343.wav (file) [Lost connection (fd:63: hGetChar: illegal operation (handle is closed))
transfer failed]
get(ok): recordings/raw/e20121106_140637_008066.wav (file) [from origin...]
get(ok): recordings/raw/e20120918_153528_008343.wav (file) [from origin...]
get(ok): recordings/raw/e20120713_153050_008015.wav (file) [from origin...]
get(ok): recordings/raw/e20120918_153634_008015.wav (file) [from origin...]
get(ok): recordings/raw/e20130322_154609_008015.wav (file) [from origin...]
get(ok): recordings/raw/e20130403_141125_008343.wav (file) [Lost connection (fd:74: hGetChar: illegal operation (handle is closed))
transfer failed]
get(ok): recordings/raw/e20130418_152236_008344.wav (file) [Lost connection (fd:75: hGetChar: illegal operation (handle is closed))
transfer failed]
get(ok): recordings/raw/e20130318_200156_008343.wav (file) [Lost connection (fd:78: hGetChar: illegal operation (handle is closed))
transfer failed]
get(ok): recordings/raw/e20130826_015914_008015.wav (file) [from origin...]
get(ok): recordings/raw/e20130621_182206_008343.wav (file) [Lost connection (fd:95: hGetChar: illegal operation (handle is closed))
transfer failed]
get(ok): recordings/raw/e20130604_135320_008109.wav (file) [from origin...]
get(ok): recordings/raw/e20130403_173734_008066.wav (file) [from origin...]
get(ok): recordings/raw/e20130708_155549_008066.wav (file) [from origin...]
get(ok): recordings/raw/e20131024_143023_008066.wav (file) [from origin...]
get(ok): recordings/raw/e20130314_125532_008066.wav (file) [from origin...]
get(ok): recordings/raw/e20130319_132608_008344.wav (file) [from origin...]
get(ok): recordings/raw/e20130314_125307_008344.wav (file) [from origin...]
get(ok): recordings/raw/e20130328_122656_008015.wav (file) [from origin...]
get(ok): recordings/raw/e20121106_140446_008344.wav (file) [from origin...]
get(ok): recordings/raw/e20130611_164617_008109.wav (file) [from origin...]
get(ok): recordings/raw/e20130329_170857_008343.wav (file) [from origin...]
get(ok): recordings/raw/e20131025_085044_008109.wav (file) [from origin...]
get(ok): recordings/raw/e20130329_134849_008109.wav (file) [from origin...]
get(ok): recordings/raw/e20130603_094158_008015.wav (file) [from origin...]
get(ok): recordings/raw/e20130612_122213_008015.wav (file) [Lost connection (fd:89: hGetChar: illegal operation (handle is closed))
transfer failed]
get(ok): recordings/raw/e20131025_085524_008344.wav (file) [from origin...]
get(ok): recordings/raw/e20130603_094347_008066.wav (file) [Lost connection (fd:86: hGetChar: illegal operation (handle is closed))
transfer failed]
get(ok): recordings/raw/e20130817_183307_008066.wav (file) [from origin...]
get(ok): recordings/raw/e20130625_131141_008344.wav (file) [from origin...]
get(ok): recordings/raw/e20130825_190818_008344.wav (file) [from origin...]
get(ok): recordings/raw/e20130612_145150_008066.wav (file) [from origin...]
get(ok): recordings/raw/e20131025_085228_008015.wav (file) [from origin...]
get(ok): recordings/raw/e20130907_113340_008343.wav (file) [from origin...]
get(ok): recordings/raw/e20130907_141536_008109.wav (file) [from origin...]
get(ok): recordings/raw/e20130708_121751_008015.wav (file) [from origin...]
get(ok): recordings/raw (directory)
action summary:
  get (ok: 39)


There are 48 recs total, looks like 39 were successfully downloaded, and 9 failed. And yet, relaunching the same command doesn't do anything. That is, no error, but also no success output:
>(ChildProjectVenv) (base) [acristia@oberon lyon]$ datalad get recordings/raw/ -J 8
(ChildProjectVenv) (base) [acristia@oberon lyon]$ 


Just in case, I tried the next step: `child-project process --threads 8 . standard basic  --format=wav --sampling=16000 --codec=pcm_s16le`

>b'./recordings/raw/e20130612_122213_008015_1: No such file or directory\n'
b'./recordings/raw/e20120724_104928_008343_1: No such file or directory\n'
b'./recordings/raw/e20130318_200156_008343_1: No such file or directory\n'
b'./recordings/raw/e20130621_182206_008343_3: No such file or directory\n'
b'./recordings/raw/e20120726_112748_008015_1: No such file or directory\n'
b'./recordings/raw/e20131025_085044_008109_1: No such file or directory\n'
b'./recordings/raw/e20130621_182206_008343_1: No such file or directory\n'
b'./recordings/raw/e20130329_170857_008343_1: No such file or directory\n'
b'./recordings/raw/e20130612_145150_008066_1: No such file or directory\n'
b'./recordings/raw/e20130418_152236_008344_1: No such file or directory\n'
b'./recordings/raw/e20130329_134849_008109_1: No such file or directory\n'
b'./recordings/raw/e20130314_125532_008066_1: No such file or directory\n'
b'./recordings/raw/e20131025_085524_008344_1: No such file or directory\n'
b'./recordings/raw/e20130621_182206_008343_2: No such file or directory\n'
b'./recordings/raw/e20130403_173734_008066_1: No such file or directory\n'
b'./recordings/raw/e20120724_105032_008109_1: No such file or directory\n'
b'./recordings/raw/e20130314_125307_008344_1: No such file or directory\n'
b'./recordings/raw/e20131025_085228_008015_1: No such file or directory\n'
b'./recordings/raw/e20130311_153528_008344_1: No such file or directory\n'
b'./recordings/raw/e20130311_153356_008109_1: No such file or directory\n'
b'./recordings/raw/e20130907_141536_008109_1: No such file or directory\n'
b'./recordings/raw/e20131024_143023_008066_1: No such file or directory\n'
b'./recordings/raw/e20130611_164617_008109_2: No such file or directory\n'
b'./recordings/raw/e20130319_132608_008344_1: No such file or directory\n'
b'./recordings/raw/e20130319_082323_008109_1: No such file or directory\n'
b'./recordings/raw/e20130311_153356_008109_2: No such file or directory\n'
b'./recordings/raw/e20131025_085228_008015_2: No such file or directory\n'
b'./recordings/raw/e20130318_152148_008066_1: No such file or directory\n'
b'./recordings/raw/e20130907_113340_008343_1: No such file or directory\n'
b'./recordings/raw/e20130611_164617_008109_1: No such file or directory\n'
b'./recordings/raw/e20130319_132608_008344_2: No such file or directory\n'
b'./recordings/raw/e20130708_121751_008015_1: No such file or directory\n'
b'./recordings/raw/e20120703_134402_008109_1: No such file or directory\n'
b'./recordings/raw/e20120713_153050_008015_1: No such file or directory\n'
b'./recordings/raw/e20130708_155549_008066_1: No such file or directory\n'
b'./recordings/raw/e20120910_154104_008109_2: No such file or directory\n'
b'./recordings/raw/e20120731_110146_008344_1: No such file or directory\n'
b'./recordings/raw/e20120910_153817_008066_2: No such file or directory\n'
b'./recordings/raw/e20120731_110527_008343_1: No such file or directory\n'
b'./recordings/raw/e20120723_155038_008015_1: No such file or directory\n'
b'./recordings/raw/e20120706_100058_008015_1: No such file or directory\n'
b'./recordings/raw/e20120628_200523_008109_1: No such file or directory\n'
b'./recordings/raw/e20120910_153817_008066_1: No such file or directory\n'
b'./recordings/raw/e20130603_094347_008066_1: No such file or directory\n'
b'./recordings/raw/e20130327_144244_008344_1: No such file or directory\n'
b'./recordings/raw/e20120910_154104_008109_1: No such file or directory\n'
b'./recordings/raw/e20130403_141125_008343_1: No such file or directory\n'
b'./recordings/raw/e20130825_190818_008344_1: No such file or directory\n'
b'./recordings/raw/e20130817_183307_008066_1: No such file or directory\n'
b'./recordings/raw/e20130322_154609_008015_1: No such file or directory\n'
b'./recordings/raw/e20120918_153528_008343_1: No such file or directory\n'
b'./recordings/raw/e20120918_153305_008344_1: No such file or directory\n'
b'./recordings/raw/e20130603_094158_008015_1: No such file or directory\n'
b'./recordings/raw/e20130604_135320_008109_1: No such file or directory\n'
b'./recordings/raw/e20120918_153634_008015_1: No such file or directory\n'
b'./recordings/raw/e20130826_015914_008015_1: No such file or directory\n'
b'./recordings/raw/e20120918_153122_008066_1: No such file or directory\n'
b'./recordings/raw/e20130328_122656_008015_1: No such file or directory\n'
b'./recordings/raw/e20121106_140637_008066_1: No such file or directory\n'
b'./recordings/raw/e20121106_140446_008344_1: No such file or directory\n'
b'./recordings/raw/e20130604_135718_008343_1: No such file or directory\n'
b'./recordings/raw/e20130625_131141_008344_1: No such file or directory\n'
b'./recordings/raw/e20120918_153634_008015_2: No such file or directory\n'
exported audio to ./recordings/converted/standard
exported processor parameters to ./recordings/converted/standard/parameters_20211026_092641.yml
(ChildProjectVenv) (base) [acristia@oberon lyon]$ 

The failure comes from an expection of _1 and _2. I think this means I should do the split of recs in the same session before the conversion, and not after. I'll pick this up later.

**Reply by lucasgautheron** [reply]
Yes, `child-project process` expects the audio to match the metadata. So you need to split the raw audio beforehand.
Also, I suggest to add a .wav extension to `recording_filename` in the metadata

**Reply by alecristia** [reply]
do you have any ideas of why .wav is missing in the metadata? I used metadata.py to create it

**Reply by alecristia** [reply]
I've just pushed my first attempt at split -- I think it fails because of the naming convention in metadata.py:
[script](https://gin.g-node.org/EL1000/lyon/src/newrecs/scripts/split_postfix.py)

> recordings/raw/e20120628_200523_008109_1.wav does not exist
Traceback (most recent call last):
  File "scripts/split_postfix.py", line 31, in <module>
    audio = AudioSegment.from_file(input_audio)
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/pydub/audio_segment.py", line 651, in from_file
    file, close_file = _fd_or_path_or_tempfile(file, 'rb', tempfile=False)
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/pydub/utils.py", line 60, in _fd_or_path_or_tempfile
    fd = open(fd, mode=mode)
FileNotFoundError: [Errno 2] No such file or directory: 'recordings/raw/e20120628_200523_008109_1.wav'
(ChildProjectVenv) (base) [acristia@oberon lyon]$ ls recordings/raw/
e20120628_200523_008109.wav  e20120910_154104_008109.wav  e20130318_152148_008066.wav  e20130418_152236_008344.wav  e20130708_155549_008066.wav
e20120703_134402_008109.wav  e20120918_153122_008066.wav  e20130318_200156_008343.wav  e20130603_094158_008015.wav  e20130817_183307_008066.wav
e20120706_100058_008015.wav  e20120918_153305_008344.wav  e20130319_082323_008109.wav  e20130603_094347_008066.wav  e20130825_190818_008344.wav
e20120713_153050_008015.wav  e20120918_153528_008343.wav  e20130319_132608_008344.wav  e20130604_135320_008109.wav  e20130826_015914_008015.wav
e20120723_155038_008015.wav  e20120918_153634_008015.wav  e20130322_154609_008015.wav  e20130604_135718_008343.wav  e20130907_113340_008343.wav
e20120724_104928_008343.wav  e20121106_140446_008344.wav  e20130327_144244_008344.wav  e20130611_164617_008109.wav  e20130907_141536_008109.wav
e20120724_105032_008109.wav  e20121106_140637_008066.wav  e20130328_122656_008015.wav  e20130612_122213_008015.wav  e20131024_143023_008066.wav
e20120726_112748_008015.wav  e20130311_153356_008109.wav  e20130329_134849_008109.wav  e20130612_145150_008066.wav  e20131025_085044_008109.wav
e20120731_110146_008344.wav  e20130311_153528_008344.wav  e20130329_170857_008343.wav  e20130621_182206_008343.wav  e20131025_085228_008015.wav
e20120731_110527_008343.wav  e20130314_125307_008344.wav  e20130403_141125_008343.wav  e20130625_131141_008344.wav  e20131025_085524_008344.wav
e20120910_153817_008066.wav  e20130314_125532_008066.wav  e20130403_173734_008066.wav  e20130708_121751_008015.wav


It'll be easier for you to troubleshoot if I move files into the `projects` section. I'm onto that.

I haven't looked at the metadata.py documentation in detail -- which I probably should too.

**Reply by alecristia** [reply]
actually, before I break anything -- is it ok if I change permissions, then move it, or will that break anything?

```
chgrp -R aclew_team EL1000
mv EL1000 /scratch1/projects/.
chmod -R g+rwx /scratch1/projects/EL1000
```

**Reply by lucasgautheron** [reply]
> do you have any ideas of why .wav is missing in the metadata? I used metadata.py to create it

Because initially EL1000 did not have audio at all, and therefore was actually an ID rather than the name of an actual file.
We can change that! There are several ways to do it.

**Comment by alecristia** [comment]
done:
- minor tweaks to metadata.py + splitting -- now validate passes the .its with just warnings (the unsplit recordings, which are inside the raw/ folder)
- failed attempt at converting: ` sbatch --mem=64G --time=5:00:00 --cpus-per-task=4 --ntasks=1 -o converting.txt child-project process . standard basic --format=wav --sampling=16000 --codec=pcm_s16le --recordings recordings/split/*`

> Traceback (most recent call last):
  File "/var/spool/slurmd/job232195/slurm_script", line 8, in <module>
    sys.exit(main())
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/ChildProject/cmdline.py", line 451, in main
    args.func(args)
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/ChildProject/cmdline.py", line 40, in <lambda>
    _parser.set_defaults(func=lambda args: cls().run(**vars(args)))
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/ChildProject/pipelines/processors.py", line 495, in run
    proc.process(f"parameters_{date}.yml")
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/ChildProject/pipelines/processors.py", line 97, in process
    self.converted = pd.concat(self.converted)
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/pandas/core/reshape/concat.py", line 284, in concat
    sort=sort,
  File "/scratch1/home/acristia/ChildProjectVenv/lib/python3.7/site-packages/pandas/core/reshape/concat.py", line 331, in __init__
    raise ValueError("No objects to concatenate")
ValueError: No objects to concatenate

**Reply by lucasgautheron** [reply]
Remove this part and try again: `--recordings recordings/split/*`

(`--recordings` sets a whitelist of raw recordings that should be processed. You don't need to do that here since you want to process everything. Also, since `recordings/split` does not exist, there will be no match and therefore an empty white-list is set)

**Reply by lucasgautheron** [reply]
Now I see that you have files in ``recordings/split``  :  how did these files end up there ? they should be in `recordings/raw`
can you `git mv` them into `recordings/raw` ?

**Reply by alecristia** [reply]
okay, my confusion came from the fact that I forgot how the convertor worked -- I now re-read the instructions [here](https://childproject.readthedocs.io/en/latest/processors.html#basic-audio-conversion)

What I had forgotten is that you typically do not need to specify a path to the recordings, and it will assume that recordings are in `recordings/raw`. I had mistakenly put the LENA outputted recordings there, and the result of splitting in `recordings/split`. I changed the structure now, such that the LENA outputted recordings are in `recordings/lena_outputted`, and the result of the split is in `recordings/raw`. I then tried: `sbatch --mem=64G --time=5:00:00 --cpus-per-task=4 --ntasks=1 -o converting.txt child-project process . standard basic --format=wav --sampling=16000 --codec=pcm_s16le` and still got the same error, no objects to concatenate.

**Reply by lucasgautheron** [reply]
I can't reproduce this error. Can you save the current state of your dataset, publish it to GIN, and maybe move your local copy in a location where I can also have access to it? (maybe you did already, but I can't find it)
